File: go/isogram/isogram.go
// Package isogram has one function to validate isograms.
package isogram

import "strings"

// IsIsogram checks if given string is an isogram.
func IsIsogram(s string) bool {
	ss := strings.ToLower(s)
	for _, c := range ss {
		if c == '-' || c == ' ' {
			continue
		}
		if strings.Count(ss, string(c)) > 1 {
			return false
		}
	}
	return true
}

File: c/isogram/isogram.c
#include "isogram.h"
#include <stdio.h>

bool is_isogram(const char *phrase)
{
    if (!phrase)
    {
        return false;
    }

    int repeated = 0;
    while (*phrase != '\0')
    {
        char c = *phrase;
        if (c == ' ' || c == '-')
        {
            phrase++;
            continue;
        }
        if (repeated & (1 << c))
        {
            return false;
        }
        repeated |= (1 << c);
        phrase++;
    }
    return true;
}
